shared/utils/TasksProvider/TasksProvider.tsx

```diff
@@ -106,11 +96,11 @@ function TasksProvider({ children }: IProps) {
       refetch();
 
       const updatedTasks = getUpdatedTasks();
-      updatedTasks && setTasks(updatedTasks);
-
-      const prevHistory = history.slice(0, currStateIndex + 1);
-      updatedTasks && setHistory([...prevHistory, [...updatedTasks]]);
-      setCurrStateIndex(prevHistory.length);
+      if (updatedTasks) {
+        setTasks(updatedTasks);
+        const storedTasksInUndo = [...undoStack];
```

Comment:
These const variables are redundant, you should be able to spread undoStack below.